addons/account/tests/test_invoice_tax_totals.py

```diff
@@ -732,3 +732,80 @@ def run_case(rounding_line, lines, expected_tax_group_amounts):
         ]
         run_case('round_per_line', lines, [16.60])
         run_case('round_globally', lines, [16.59])
+
+    def test_cash_rounding_amount_total_rounded(self):
+        tax_15 = self.env['account.tax'].create({
+            'name': "tax_15",
+            'amount_type': 'percent',
+            'amount': 15.0,
+        })
+        cash_rounding = self.env['account.cash.rounding'].create({
+            'name': 'Rounding HALF-UP',
+            'rounding': 1,
+            'strategy': 'biggest_tax',
+            'rounding_method': 'HALF-UP',
+        })
+
+        invoice = self.init_invoice('out_invoice', amounts=[378], taxes=tax_15)
+        invoice.invoice_cash_rounding_id = cash_rounding
+        self.assertEqual(invoice.tax_totals['amount_total_rounded'], 435)
+
+        bill = self.init_invoice('in_invoice', amounts=[378], taxes=tax_15)
+        bill.invoice_cash_rounding_id = cash_rounding
+        self.assertEqual(bill.tax_totals['amount_total_rounded'], 435)
+
+    def test_cash_rounding_amount_total_rounded_foreign_currency(self):
+        tax_15 = self.env['account.tax'].create({
+            'name': "tax_15",
+            'amount_type': 'percent',
+            'amount': 15.0,
+        })
+        cash_rounding = self.env['account.cash.rounding'].create({
+            'name': 'Rounding HALF-UP',
+            'rounding': 10,
+            'strategy': 'biggest_tax',
+            'rounding_method': 'HALF-UP',
+        })
+        self.env['res.currency.rate'].create({
+            'name': '2023-01-01',
+            'rate': 0.2,
+            'currency_id': self.currency_data['currency'].id,
+            'company_id': self.env.company.id,
+        })
+        invoice = self.env['account.move'].create({
+            'move_type': 'out_invoice',
+            'partner_id': self.partner_a.id,
+            'invoice_date': '2023-01-01',
+            'currency_id': self.currency_data['currency'].id,
+            'invoice_line_ids': [
+                Command.create({
+                    'name': 'line',
+                    'display_type': 'product',
+                    'account_id': self.company_data['default_account_revenue'].id,
+                    'price_unit': 100,
+                    'tax_ids': [tax_15.id],
+                })
+            ]
+        })
+        invoice.invoice_cash_rounding_id = cash_rounding
+        self.assertEqual(invoice.tax_totals['amount_total'], 115)
+        self.assertEqual(invoice.tax_totals['amount_total_rounded'], 120)
+
+        bill = self.env['account.move'].create({
```

Comment:
Maybe loop on the move types in order to [keep it DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)